### PR TITLE
Create ForeignKey earlier in low downtime migration

### PR DESCRIPTION
**Pull request by aksharauke**
During low downtime migration - In the Create database - also enable the foreign key creation
During low downtime migration - If the database already exists and there are foreign keys to be applied - give an exception. Since applying foreign keys on existing database takes long time, even if tables are empty. 

Note that for PostgreSQL dialect, applying DDL as part of database creation is not supported, hence the behavior for PostgreSQL dialect will remain like current - the foreign keys will be applied once the Dataflow jobs start

### CODE REVIEW

**Comment by aksharauke** [comment]
#review @shreyakhajanchi  @manitgupta

File: conversion/conversion.go [hunk 1]
```diff
@@ -676,12 +676,15 @@ func CreateOrUpdateDatabase(ctx context.Context, adminClient *database.DatabaseA
 		ctx = metadata.AppendToOutgoingContext(ctx, constants.MigrationMetadataKey, migrationMetadataValue)
 	}
 	if dbExists {
+		if conv.SpDialect != constants.DIALECT_POSTGRESQL && migrationType == constants.DATAFLOW_MIGRATION {
+			return fmt.Errorf("Low downtime schema migration cannot work on existing database")
```
Comment (manitgupta) [inline conversion/conversion.go:680]: Lets reword this to "Harbourbridge does not support minimal downtime schema/schema-and-data migrations to an existing database".

**Review by manitgupta — APPROVED** [review]

File: conversion/conversion.go (hunk 1 above)
Comment (aksharauke) [reply conversion/conversion.go]: Done

**Review by shreyakhajanchi — APPROVED** [review]
LGTM
